File: web/tests/reference/prefix-sum/contiguous-array.py
def find_max_length(nums):
    first_seen = {0: -1}
    running = 0
    best = 0
    for i, x in enumerate(nums):
        running += 1 if x == 1 else -1
        if running in first_seen:
            best = max(best, i - first_seen[running])
        else:
            first_seen[running] = i
    return best

File: web/tests/reference/prefix-sum/contiguous-array.js
function findMaxLength(nums) {
  const firstSeen = new Map([[0, -1]]);
  let running = 0;
  let best = 0;
  for (let i = 0; i < nums.length; i++) {
    running += nums[i] === 1 ? 1 : -1;
    if (firstSeen.has(running)) best = Math.max(best, i - firstSeen.get(running));
    else firstSeen.set(running, i);
  }
  return best;
}
